source/postprocess/crystal_preferred_orientation.cc

```diff
@@ -0,0 +1,816 @@
+                             "each MPI process. It can write out the following data: "
+                             "olivine volume fraction, olivine A matrix, olivine Euler angles, "
+                             "enstatite volume fraction, enstatite A matrix, enstatite Euler angles. \n"
+                             "Note that the A matrix and Euler angles both contain the same "
+                             "information, but in a different format. Euler angles are recommended "
+                             "over the A matrix since they only require to write 3 values instead "
+                             "of 9. If the list is empty, this file will not be written. "
+                             "Furthermore, the entries will be written out in the order given, "
+                             "and if entries are entered muliple times, they will be written "
+                             "out multiple times.");
+          prm.declare_entry ("Compress cpo data files", "true",
+                             Patterns::Bool(),
+                             "Wether to compress the raw and weighted cpo data output files with zlib.");
+        }
+        prm.leave_subsection ();
+      }
+      prm.leave_subsection ();
+
+    }
+
+
+    template <int dim>
+    void
+    CrystalPreferredOrientation<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      end_time = prm.get_double ("End time");
+      if (this->convert_output_to_years())
+        end_time *= year_in_seconds;
+
+      unsigned int n_minerals;
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Particles");
+        {
+          prm.enter_subsection("Crystal Preferred Orientation");
+          {
+            prm.enter_subsection("Initial grains");
+            {
+              // Static variable of CPO has not been initialize yet, so we need to get it directly.
+              const std::vector<std::string> temp_deformation_type_selector = dealii::Utilities::split_string_list(prm.get("Minerals"));
+              n_minerals = temp_deformation_type_selector.size();
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection ();
+        }
+        prm.leave_subsection ();
+        prm.enter_subsection("Crystal Preferred Orientation");
+        {
+          output_interval = prm.get_double ("Time between data output");
+          if (this->convert_output_to_years())
+            output_interval *= year_in_seconds;
+
+          random_number_seed = prm.get_integer ("Random number seed");
+
+          //AssertThrow(this->get_parameters().run_postprocessors_on_nonlinear_iterations == false,
+          //            ExcMessage("Postprocessing nonlinear iterations in models with "
+          //                       "particles is currently not supported."));
+
+          aspect::Utilities::create_directory (this->get_output_directory() + "particle_CPO/",
```

Comment:
stay consistent with the existing directory name

```suggestion
          aspect::Utilities::create_directory (this->get_output_directory() + "particles_CPO/",
```

But ideally I would prefer an even more expressive name. Something like `particles_cpo_properties`?